apps/fishing-map/features/analysis/AnalysisPeriodComparison.tsx

```diff
@@ -37,6 +39,58 @@ const AnalysisPeriodComparison: React.FC<AnalysisTypeProps> = (props) => {
     MAX_DATE,
   } = useAnalysisTimeCompareConnect('periodComparison')
 
+  const trackAndChangeComparisonDate = useCallback((date) => {
+    uaEvent({
+      category: 'Analysis',
+      action: `Select comparison date in 'period comparison'`,
+      label: JSON.stringify({
+        date: date.target.value,
+        regionName: analysisAreaName,
+        sourceNames: dataviews.flatMap(dataview => getSourcesSelectedInDataview(dataview).map(source => source.label))
+      })
+    })
+    onCompareStartChange(date)
+  }, [onCompareStartChange])
+
+  const trackAndChangeBaselineDate = useCallback((date) => {
+    uaEvent({
+      category: 'Analysis',
+      action: `Select baseline date in 'period comparison'`,
+      label: JSON.stringify({
+        date: date.target.value,
+        regionName: analysisAreaName,
+        sourceNames: dataviews.flatMap(dataview => getSourcesSelectedInDataview(dataview).map(source => source.label))
+      })
+    })
+    onCompareStartChange(date)
```

Comment:
I think you should be calling `onStartChange` here, look at this line: https://github.com/GlobalFishingWatch/frontend/pull/1204/files#diff-63134b5729019ef9b974a721d2a391154fc931c4210efad2c398264111284229L101